### ISSUE TITLE
Example inaccurate for pytorch lightning

### ISSUE THREAD

**Issue by ysyyork** [horovod/horovod#3329]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) Pytorch
2. Framework version: 1.10.1
3. Horovod version: 0.23.0
4. MPI version: 3.0.0
5. CUDA version: 11.1
6. NCCL version: 2.7.8
7. Python version: 3.7.11
8. Spark / PySpark version:
9. Ray version:
10. OS and version:
11. GCC version:
12. CMake version:

The example for pytorch lightning for mnist is inaccurate. For [this line](https://github.com/horovod/horovod/blob/8d34c85ce7ec76e81fb3be99418b0e4d35204dc3/examples/pytorch/pytorch_lightning_mnist.py#L56)
```
    def configure_optimizers(self):
        return optim.SGD(self.parameters(), lr=0.01, momentum=0.5)
```
It uses a torch optimizer. In the framework, this optimizer would be converted to a horovod distributed optimizer automatically. So, this example is working fine. However, if I add any lr_scheduler here, then it becomes tricky. For example, if we apply a `ReduceLROnPlatue` scheduler, the code would become like below:
```
def configure_optimizers(self):
        optimizer = optim.SGD(self.parameters(), lr=0.01, momentum=0.5)
        lr_scheduler = ReduceLROnPlateau(optimizer, factor=0.5, patience=10)
        return {
                'optimizer': optimizer,
                'lr_scheduler': {
                    'scheduler': lr_scheduler,
                    'monitor': 'val_loss'
                }
        }
```
Then during training, the scheduler will keep on optimizing the torch optimizer but not the horovod distributed one. The distributed one would stay same forever. I think this is because the auto conversion between torch opt and hvd opt is down after the `configure_optimizers` function returned which is after the `lr_scheduler` is initiated. 

If you add a line `optimizer = hvd.optimizer.DistributedOptimizer(optimizer)` before lr_scheduler initialization, it starts to work. I didn't dig deep into the source but I guess the auto conversion will check if the optimizer is already a hvd one or not, if yes, then it will just use it as is. 

I think it's worth some explanation or some editing in this example to illustrate this. Otherwise it's very hard to debug.

**Comment by chongxiaoc** [comment]
@irasit
